metadata-ingestion/src/datahub/ingestion/source/sigma/sigma_api.py

```diff
@@ -20,7 +20,9 @@
 class SigmaAPI:
     def __init__(self, config: SigmaSourceConfig) -> None:
         self.config = config
-        self.users: Dict = {}
+        self.workspaces: Dict[str, Workspace] = {}
+        self.users: Dict[str, str] = {}
+        self.datasets: Dict[str, str] = {}
```

Comment:
why are we saving these? I would've expected these to be saved in sigma.py, not in sigma_api.py

Reply:
Removed `self.datasets` but others can't be moved or removed. We even can't use `functools.lru_cache` as it gives problem with function fetching response from GET API.